MetricsDashboard.DataAccess/Entities/Metric.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using MetricsDashboard.Dto;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace MetricsDashboard.DataAccess.Entities
+{
+    public class Metric<TValue> : IMetric
+    {
+        [BsonId]
+        [BsonRepresentation(BsonType.ObjectId)]
+        public string Id { get; set; }
+
+        [BsonElement("metricName")]
+        public string MetricName { get; set; }
+
+        [BsonElement("metricKind")]
+        public string MetricKind { get; set; }
+
+        [BsonElement("addedOn")]
+        public DateTimeOffset AddedOn { get; set; }
+
+        [BsonElement("value")]
+        public TValue Value { get; set; }
+
+        public Metric ToDto() => new Metric { AddedOn = AddedOn, Value = Value };
```

Comment:
Entity should't have information about DTO object. Some service or repository can have knowledge how to convert Entity to Dto.

Reply:
Created service and implemented mapping from Entity to DTO, removed DTO knowledge from Entites.